host/integrationbase.go

```diff
@@ -267,6 +227,12 @@ func (s *IntegrationBase) registerDomain(
 	})
 }
 
+func (s *IntegrationBase) domainCacheRefresh() {
+	s.testClusterConfig.TimeSource.Advance(cache.DomainCacheRefreshInterval + time.Second)
+	// this sleep is necessary to yield execution to other goroutines. not 100% guaranteed to work
+	time.Sleep(2 * time.Second)
```

Comment:
https://pkg.go.dev/runtime#Gosched
followed by the time.Sleep might work

Reply:
Tried and it failed. Even though execution is yielded the test goroutine is still in the list of available goroutines to resume immediately after. Hence `Gosched` approach doesn't guarantee the test execution is not resumed until other pending goroutines have a chance to resume. `time.Sleep` also doesn't guarantee that but reduces the chances drastically.